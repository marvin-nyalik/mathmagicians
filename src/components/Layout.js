import Navigation from './Navigation';

const layout = ({ children }) => (
  <div className="container">
    <Navigation />
    {children}
  </div>
);
export default layout;
