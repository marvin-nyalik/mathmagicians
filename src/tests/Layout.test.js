import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom'; // Import MemoryRouter

import Layout from '../components/Layout';

test('renders Layout component correctly', () => {
  const { container } = render(
    <MemoryRouter>
      {/* Wrap the Layout component with MemoryRouter */}
      <Layout />
    </MemoryRouter>,
  );
  expect(container.firstChild).toMatchSnapshot();
});
