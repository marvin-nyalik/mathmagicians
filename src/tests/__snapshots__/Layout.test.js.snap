// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Layout component correctly 1`] = `
<div
  class="container"
>
  <nav
    class="nav"
  >
    <h2>
      <a
        aria-current="page"
        class="logo active"
        href="/"
      >
        Math-Magicians
      </a>
    </h2>
    <ul
      class="ul"
    >
      <li>
        <a
          aria-current="page"
          class="navlink active"
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          class="navlink"
          href="/calculator"
        >
          Calculate
        </a>
      </li>
      <li>
        <a
          class="navlink"
          href="/quote"
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
